cnf-certification-test/accesscontrol/suite.go

```diff
@@ -303,3 +322,132 @@ func TestPodServiceAccount(env *provider.TestEnvironment) {
 		ginkgo.Fail(fmt.Sprintf("%d pods don't have a service account name.", n))
 	}
 }
+
+// TestPodRoleBindings verifies that the pod utilizes a valid role binding that does not cross namespaces
+//nolint:dupl
+func TestPodRoleBindings(env *provider.TestEnvironment) {
+	ginkgo.By("Should not have RoleBinding in other namespaces")
+	failedPods := []string{}
+
+	for _, put := range env.Pods {
+		ginkgo.By(fmt.Sprintf("Testing role binding for pod: %s namespace: %s", put.Name, put.Namespace))
+		if put.Spec.ServiceAccountName == "" {
+			ginkgo.Skip("Can not test when serviceAccountName is empty. Please check previous tests for failures")
+		}
+
+		// Create a new object with the ability to gather rolebinding specs.
+		rbTester := rbac.NewRoleBindingTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+
+		// Get any rolebindings that do not belong to the pod namespace.
+		roleBindings, err := rbTester.GetRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(roleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following role bindings: %s", put.Name, roleBindings)
+			failedPods = append(failedPods, put.Name)
+		}
+	}
+	if n := len(failedPods); n > 0 {
+		logrus.Debugf("Pods with role bindings: %+v", failedPods)
+		ginkgo.Fail(fmt.Sprintf("%d pods have role bindings in other namespaces.", n))
+	}
+}
+
+// TestPodClusterRoleBindings verifies that the pod utilizes a valid cluster role binding that does not cross namespaces
+//nolint:dupl
+func TestPodClusterRoleBindings(env *provider.TestEnvironment) {
+	ginkgo.By("Should not have ClusterRoleBinding in other namespaces")
+	failedPods := []string{}
+
+	for _, put := range env.Pods {
+		ginkgo.By(fmt.Sprintf("Testing cluster role binding for pod: %s namespace: %s", put.Name, put.Namespace))
+		if put.Spec.ServiceAccountName == "" {
+			ginkgo.Skip("Can not test when serviceAccountName is empty. Please check previous tests for failures")
+		}
+
+		// Create a new object with the ability to gather clusterrolebinding specs.
+		rbTester := rbac.NewClusterRoleBindingTester(put.Spec.ServiceAccountName, put.Namespace, clientsholder.NewClientsHolder())
+
+		// Get any clusterrolebindings that do not belong to the pod namespace.
+		clusterRoleBindings, err := rbTester.GetClusterRoleBindings()
+		if err != nil {
+			failedPods = append(failedPods, put.Name)
+		}
+
+		if len(clusterRoleBindings) > 0 {
+			logrus.Warnf("Pod: %s has the following cluster role bindings: %s", put.Name, clusterRoleBindings)
```

Comment:
Add failing clusterRoleBindings to the claims file

Reply:
Done.